### PR TITLE
Add resilience to WOF imports

### PR DESCRIPTION
**Pull request by rmarianski**
Connects with: https://github.com/mapzen/vector-datasource/issues/393

This isn't finished yet, but the basic structure of how to handle failures should be in place.

@zerebubuth: could you take a quick look and let me know if I went down the wrong path, or if you have any suggestions thus far?

### CODE REVIEW

File: tilequeue/wof.py [hunk 1]
```diff
@@ -689,30 +799,38 @@ def fetch_toi():
...
+            failed_wof_ids.add(failure_wof_id)
+            ids_to_add.discard(failure_wof_id)
+            ids_to_update.discard(failure_wof_id)
+
+        # now we'll want to ensure that these ids are not present in the diffs
+        new_diffs = []
+        for n1, n2 in diffs:
+            if n2 is not None and n2.wof_id in failed_wof_ids:
+                continue
+            new_diffs.append((n1, n2))
```
Comment (zerebubuth) [inline tilequeue/wof.py]: Might this be easier to understand if it were?

```python
if n2 is None or n2.wof_id not in failed_wof_ids:
    new_diffs.append((n1, n2))
```

File: tilequeue/wof.py [hunk 2]
```diff
@@ -689,30 +799,38 @@ def fetch_toi():
...
-        # once we detect fetch failures, and I think it's best to
-        # handle all these cases at once, but just making a note in
-        # here now to serve as a reminder
+        # we should just remove any neighbourhoods from add/update lists
+        # also keep track of these ids to remove from the diffs too
+        failed_wof_ids = set()
+        for failure in failures:
+            failure_wof_id = failure.wof_id
+            # TODO logging funky failures is noisy
+            # maybe we just log the count of funky failures?
```
Comment (zerebubuth) [inline tilequeue/wof.py]: I would agree - if we're expecting that we might get quite a lot of these in the future, to log them every hour is just going to bloat the logs. Does this mean that `NeighbourhoodFailure` needs a flag for funkiness?

File: tilequeue/wof.py (hunk 1 above)
Comment (rmarianski) [inline tilequeue/wof.py]: Agreed, updated in e771584.

File: tilequeue/wof.py (hunk 2 above)
Comment (rmarianski) [inline tilequeue/wof.py]: Yup, added that and suppressed individual logging for funky records in 3f5de44.

**Comment by zerebubuth** [comment]
:+1: looks good to me

**Pull request merged**